chain/network/src/peer_manager.rs

```diff
@@ -1679,6 +2123,95 @@ impl Handler<InboundTcpConnect> for PeerManagerActor {
     }
 }
 
+#[cfg(feature = "adversarial")]
+#[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+impl Handler<StartRoutingTableSyncRequest> for PeerManagerActor {
+    type Result = ();
+
+    #[perf]
+    fn handle(&mut self, msg: StartRoutingTableSyncRequest, ctx: &mut Self::Context) {
+        if let Some(active_peer) = self.active_peers.get(&msg.peer_id) {
+            let addr = active_peer.addr.clone();
+            self.initialize_routing_table_exchange(msg.peer_id, PeerType::Inbound, addr, ctx);
+        }
+    }
+}
+
+#[cfg(feature = "adversarial")]
+impl Handler<SetAdvOptions> for PeerManagerActor {
+    type Result = SetAdvOptionsResult;
+
+    #[perf]
+    fn handle(&mut self, msg: SetAdvOptions, _ctx: &mut Self::Context) -> SetAdvOptionsResult {
+        if let Some(disable_edge_propagation) = msg.disable_edge_propagation {
+            self.adv_disable_edge_propagation = disable_edge_propagation;
+        }
+        if let Some(disable_edge_signature_verification) = msg.disable_edge_signature_verification {
+            self.adv_disable_edge_signature_verification = disable_edge_signature_verification;
+        }
+        if let Some(disable_edge_pruning) = msg.disable_edge_pruning {
+            self.adv_disable_edge_pruning = disable_edge_pruning;
+        }
+        SetAdvOptionsResult {}
+    }
+}
+
+impl Handler<GetRoutingTable> for PeerManagerActor {
+    type Result = GetRoutingTableResult;
+
+    #[perf]
+    fn handle(&mut self, msg: GetRoutingTable, _ctx: &mut Self::Context) -> GetRoutingTableResult {
+        GetRoutingTableResult {
+            edges_info: self
+                .routing_table
+                .edges_info
+                .iter()
+                .map(|(_, e)| (SimpleEdge::new(e.peer0.clone(), e.peer1.clone(), e.nonce)))
+                .collect(),
+        }
+    }
+}
+
+#[cfg(feature = "adversarial")]
+impl SetRoutingTableRequest {
+    pub fn parse(value: Option<Value>) -> Result<Self, RpcError> {
+        if let Some(value) = value {
+            serde_json::from_value(value)
+                .map_err(|err| RpcError(format!("Failed parsing args: {}", err)))
```

Comment:
Do you feel it is the right place to parse the thing? Is jsonrpc the only place where we handle it as JSON?

Reply:
removed